### ISSUE TITLE
Add TemporalClient.BuildActivityStub<T>() and BuildWorkflowStub<T>()

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#796]
It can take some time to compile the generated workflow and activity stubs just before we register workflow/activity workers.  Neon.Temporal unit tests need to register something like 130+ workflows (which is a lot) and this process takes maybe 30 seconds to complete.  This acts to delay the worker becoming active.

Currently, we are compiling each generated stub one at a time into its own in-memory assembly and then loading the generated assembly.  It appears that cranking up the C# compiler is taking a half second or so to compile each of these stubs.

We could potentially improve the performance by generating one large source file for all of the workflows and activities being registered and compiling them all at once into a single assembly.

Probably a **low priority** though since most workers probably won't have nearly this many workflow/activity implementations.

**Comment by jefflill** [comment]
We're currently generating workflow and activity stubs for each implementation when the worker starts.  The idea being that it takes maybe 500ms to compile each stub and if a workflow created several stubs at once, compiling them all may take longer than the decision task timeout, failing the workflow.  The idea of compiling stubs for all of the workflow/activities implementations at workflow start was a primitive way of avoiding this.

The problem is that this is slow and there's no guarantee that the workflows aren't going to be calling (and needing to compile) stubs for workflows/activities implemented elsewhere, so this isn't really a comprehensive fix anyway.

This is never going to be a problem for most workflows anyway because stub generation is relatively quick and most workflows will problem only need a few stubs and because stub types are cached and will only need to be compiled once for each worker process.

A better approach will be to allow the user to precompile specific stubs if this is a actually problem.

**Comment by jefflill** [comment]
Apparently, I never completed this.  Let's make this change to **Neon.Cadence** as well and also add methods that create stubs for an entire assembly.

**Comment by jefflill** [comment]
@johncburns1: I'm adding three static `CadenceClient` and `TemporalClient` methods users can use to mitigate decision task timeouts by prebuilding the backing stub classes before starting any workers:
```
void BuildActivityStub<TActivity>() where TActivity : IActivity;
void BuildWorkflowStub<TWorkflow>() where TWorkflow : IWorkflow;
void BuildAssemblyStubs<TWorkflow>() where TWorkflow : IWorkflow;
```

**Comment by jefflill** [comment]
**DONE** for v3.0.0

The nice thing about this change for we Cadence/Temporal developers is that we'll no longer have to wait the 30+ seconds it takes to prebuild all of the unit test stubs when we're debugging a single workflow/activity problem.
